src/request/caches.rs

```diff
@@ -42,6 +43,27 @@ impl Default for LocalCache {
     }
 }
 
+#[derive(Debug)]
+enum Action {
+    Clear,
+    Config,
+    Keys,
+    Size,
+    Stats,
+}
+
+impl fmt::Display for Action {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
```

Comment:
This is just delegating to the `Debug` impl. You might want to either move callers to use the debug impl directly or use a "nicer"/non-Rusty output (ie. `clear`, `config`, `size`, etc)... unless you really prefer the `Debug` output.

Reply:
The method returns `Clear`, `Config`, etc. and then in the caller I just use `to_string().to_lowercase()`. That's needed because query params in infinispan are case-sensitive.

Taking that into account, do you see a way yo simplify this?
```rust
#[derive(Debug)]
enum Action {
    Clear,
    Config,
    Keys,
    Size,
    Stats,
}

impl fmt::Display for Action {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "{:?}", self)
    }
}

impl Action {
    pub fn to_query_args(&self) -> String {
        format!("action={}", self.to_string().to_lowercase())
    }
}

```

Reply:
> Taking that into account, do you see a way yo simplify this?
> 
> ```rust
> #[derive(Debug)]
> enum Action {
>     Clear,
>     Config,
>     Keys,
>     Size,
>     Stats,
> }
> 
> impl fmt::Display for Action {
>     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
>         write!(f, "{:?}", self)
>     }
> }
> 
> impl Action {
>     pub fn to_query_args(&self) -> String {
>         format!("action={}", self.to_string().to_lowercase())
>     }
> }
> ```

Yes, you are creating a heap-allocated `String` with the `ToString` impl used by `Display`. Then you are allocating a new `String` and copying the `ToString`-provided one over to this new one, where you'll lowercase some characters, and then you'll deallocate the `ToString` originated `String`. You can avoid this extra allocation/deallocation + copying by using `core::str::make_ascii_lowercase`, since you don't care about potentially not lowercasing UTF-8 non-ASCII characters (which could potentially not allow in-place operation because some characters might take more than one byte).

Reply:
Right, I've simplified this :+1: